### PR TITLE
Add version in store's meta information 

### PR DESCRIPTION
**Pull request by nolouch**
For compatibility upgrade.

### CODE REVIEW

File: proto/pdpb.proto [hunk 1]
```diff
@@ -53,12 +53,15 @@ service PD {
 message RequestHeader {
     // cluster_id is the ID of the cluster which be sent to.
     uint64 cluster_id = 1;
+    string version = 2;
+    string require_min_version = 3;
```
Comment (disksing) [inline proto/pdpb.proto]: I think we need some comments here.

**Comment by siddontang** [comment]
I prefer using number (major, minor, patch, unstable) for version, not string

Btw, containing the version in every request is heavy?

**Comment by disksing** [comment]
@siddontang Differs from your point of view, I think string is better. It's more readable, easier to construct a message, and more convenient to compare by using appropriate package (e.g. https://github.com/blang/semver)

**Comment by Hoverbear** [comment]
@disksing If we're serializing this onto every request I would be interested if there is a size difference between choosing string or a `(u8,u8,u8,u8)` as @siddontang suggested.

**Comment by disksing** [comment]
@Hoverbear I don't think the difference will be too significant. Notice that protobuf needs to use an extra varint to marshal each field.

**Comment by nolouch** [comment]
Actually, I compare them via:
```
	h := &pdpb.RequestHeader{
		ClusterId:         6571635133475700175,
		Version:           "v2.0.0-rc3",
		RequireMinVersion: "v2.0.0-rc4",
	}
	b, _ := proto.Marshal(h)
	fmt.Println("RequestHeader:", len(b))
	fmt.Println("RequestHeader(gizp):", gzipLen(b))

	v := &pdpb.Version{Major: 2, Minor: 1, Patch: 1, Unstable: 1}
	v2 := &pdpb.Version{Major: 2, Minor: 1, Patch: 1, Unstable: 14}
	h2 := &pdpb.RequestHeader2{
		ClusterId:         6571635133475700175,
		Version:           v,
		RequireMinVersion: v2,
	}

	b, _ = proto.Marshal(h2)
	fmt.Println("RequestHeader(use Version):", len(b))
	fmt.Println("RequestHeader2(use Version & gizp):", gzipLen(b))
```
The Result:
```
RequestHeader: 34
RequestHeader(gizp): 56
RequestHeader2(use Version): 30
RequestHeader2(use Version & gizp): 54
```

**Comment by nolouch** [comment]
PTAL @disksing @siddontang

**Comment by disksing** [comment]
LGTM. Please use a more appropriate PR title.

**Pull request merged**